ix-dev/community/urbackup/templates/test_values/basic-values.yaml

```diff
@@ -0,0 +1,53 @@
+resources:
+  limits:
+    cpus: 2.0
+    memory: 4096
+TZ: America/Denver
+
+urbackup:
+  additional_envs: []
+network:
+  host_network: false
+  web_port:
+    bind_mode: published
+    port_number: 55414
+  backup_port_1:
+    bind_mode: published
+    port_number: 55413
+  backup_port_2:
+    bind_mode: published
+    port_number: 55415
+  backup_port_3:
+    bind_mode: published
+    port_number: 35623
+
+run_as:
+  user: 0
+  group: 0
```

Comment:
Why testing as root?

Reply:
i cant seem to make it work without root. I tested as userid 568 and added all docker capabilities but it fails trying to lock /etc/passwd

> 2025-01-24 10:04:15 urbackup-server-1  | usermod: Permission denied.
2025-01-24 10:04:15 urbackup-server-1  | usermod: cannot lock /etc/passwd; try again later.

on my non truenas server i run this application and inside the container it is setup as root

> [media@nas ~]$ docker exec -it urbackup-server-1 /bin/bash
root@cafeebfd16f5:/# id
uid=0(root) gid=0(root) groups=0(root)
root@cafeebfd16f5:/#

maybe i can mount /etc/passwd as read only. ill try

Reply:
mount /etc/passwd read only got me further

> 2025-01-24 10:16:29 urbackup-server-1  | usermod: user 'urbackup' does not exist

not sure about this one without modifying the dockerfile upstream

Reply:
https://github.com/uroni/urbackup-server-docker/blob/latest/entrypoint.sh

> Giving the user and group "urbackup" the provided UID/GID
if [[ $PUID != "" ]]
then
	usermod -u $PUID -o urbackup
else
	usermod -u 101 -o urbackup
fi
if [[ $PGID != "" ]]
then
	groupmod -g $PGID -o urbackup
else
	groupmod -g 101 -o urbackup
fi
chown urbackup:urbackup /backups
chown urbackup:urbackup /var/urbackup
exec urbackupsrv "$@"

it wants to run usermod but the user doesnt exist.. hm.

Reply:
Okay so, it runs as root but starts subprocesses as non-root.

So,
c1.set_user(0, 0)
c1.add_caps(["CHOWN", "FOWNER", "DAC_OVERRIDE", "SETUID", "SETGID"])

You still leave the run_as section as is, and it will automagically set the PUID etc.

Reply:
great. updated to include this